lib/determinator/retrieve/dynaconf.rb

```diff
@@ -0,0 +1,31 @@
+require 'faraday'
+
+module Determinator
+  module Retrieve
+    # A class which loads features from Dynaconf server
+    class Dynaconf
+      # @param :host [String] The host for local Dynaconf server
+      # @param :client [String] Faraday client instance, defaults to a new instance
+      def initialize(host:, client: default_client)
+        @host = host
+        @client = client
+      end
+
+      def retrieve(feature_id)
+        url = "http://#{@host}/scopes/florence-#{feature_id}/feature"
+
+        payload = @client.get(url).body
+        Determinator::Serializers::JSON.load(payload)
+      rescue => e
+        Determinator.notice_error(e)
+        nil
+      end
+
+      private
+
+      def default_client
```

Comment:
Is there any way you could send some form of client identifier to DynaConf? This way we may be able to for example understand which application is trying to retrieve non-existent flags.

Reply:
Determinator doesn't currently collect this information, so that'll be a little tricky; certainly a larger piece — maybe something for the next PR?

Reply:
We definitely can add `User-Agent` with something like `Determinator: 2.0.3`. 

The tricky bit is that I chose to allow the client to be injected, so that each application can configure the Faraday instance the way they want. For example in terms of retries, Datadog APM integration, etc. This is just a fallback client and it wouldn't know what application is it running on. We could check for some standard ENVs in our environment, if they exist but this gem is open source, so it doesn't feel right.

Any ideas?

Reply:
`Determinator::Retrieve::Dynaconf.new()` could accept a user-agent string or another identifier, to be used here when building the Faraday connection.

Reply:
I think this is a very mixed approach, where we allow you to inject the full client and also some parts of it to override later. 

My opinion is that there are some sensible defaults and if you inject an override, you configure it. It's a developer setting this up, not a user, so they should be sensible about what they provide.

Reply:
I think I'll consider adding `service_name` to the constructor, which will be used to enforce the `User-Agent`.

Reply:
Since injecting a pre-built Faraday client addresses this issue, please consider this comment as a non-blocking suggestion.

Personally I'm lightly biased towards opinionated interfaces with good defaults, even though this often means taking control away from the end user of the library (let's forget that this is Ruby and people can monkey patch stuff).

The first reason is consistency, because if passing a `User-Agent` is left to the users of the library, some of them will forget it. Also, if Dynaconf starts requiring some mandatory request header, then I think it would be convenient to just release new version of the library that will support that [1], without relying on the users of the library to be familiar with how the low-level plumbing of the Dynaconf API is evolving.

The second reason is that it makes the library easier to use and easier to maintain. I would ask myself the question: why does the end user need to know that this library uses Faraday? What if in the future we want to remove Faraday and use something else?

[1]: e.g. "Changelog: now supports Dynaconf 1.42".